src/sankoff_handler.cpp

```diff
@@ -3,34 +3,38 @@
 
 #include "sankoff_handler.hpp"
 
-SankoffPartialVec SankoffHandler::GenerateLeafPartials(const size_t node_count) {
-  SankoffPartialVec p_partials_left(site_pattern_.PatternCount());
-  for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
-       pattern_idx++) {
-    SankoffPartial site_partials(node_count, state_count_);
-    site_partials.setZero();
-
-    // Only the leaf partials will have the big_double_ infinity substitute.
-    site_partials.block(0, 0, site_pattern_.TaxonCount(), state_count_)
-        .fill(big_double_);
-    for (size_t seq = 0; seq < site_pattern_.TaxonCount(); seq++) {
-      auto site_val = site_pattern_.GetPatternSymbol(seq, pattern_idx);
+void SankoffHandler::GenerateLeafPartials() {
+  // first check that the psv_handler has been resized to deal with the leaf labels
+  Assert(psv_handler_.GetNodeCount() >= site_pattern_.TaxonCount(),
+		  "Error in SankoffHandler::GenerateLeafPartials: "
+		  "psv_handler_ should be initialized to accomodate" 
+		  "the number of leaf nodes in the site_pattern_.");
+
+  // Iterate over all leaf nodes to instantiate each with P partial values
+  for (size_t leaf_node = 0; leaf_node < site_pattern_.TaxonCount(); leaf_node++) {
+    SankoffPartial node_partials(state_count_, site_pattern_.PatternCount());
+    // set leaf node partial to have big_double_ infinity substitute
+    node_partials.block(0, 0, state_count_, site_pattern_.PatternCount()).fill(
+		    big_double_);
+    // now fill in appropriate entries of the leaf-partial where non-infinite
+    for (size_t pattern_idx = 0; pattern_idx < site_pattern_.PatternCount();
+         pattern_idx++) {
+      auto site_val = site_pattern_.GetPatternSymbol(leaf_node, pattern_idx);
       if (site_val < state_count_) {
-        site_partials(seq, site_val) = 0.;
+        node_partials(site_val, pattern_idx) = 0.;
       } else if (site_val == state_count_) {
         // Leaves with gaps in sequence and ambiguous nucleotides are assigned sankoff
         // partial vector [0, 0, 0, 0] at the corresponding site.
-        site_partials.block<1, state_count_>(seq, 0).fill(0);
+        node_partials.block<state_count_, 1>(0, pattern_idx).fill(0);
```

Comment:
Given that this sort of statement gets repeated a lot, can we have an attractively-named alternative with zero overhead so we could say something like `GetColumn(node_partials, pattern_idx).fill(0)`.

cc @ognian- who now knows a lot about eigen ref types.

Reply:
resolved using `Eigen::DenseBase::col`